frappe/utils/oauth.py

```diff
@@ -231,7 +231,7 @@ def update_oauth_user(user, data, provider):
 		save = True
 		user = frappe.new_doc("User")
 
-		gender = (data.get("gender") or "").title()
+		gender = (data.get("gender") or "Non-Confirming").title()
 
 		if not frappe.db.exists("Gender", gender):
```

Comment:
```suggestion
		gender = (data.get("gender") or "").title()

		if gender and not frappe.db.exists("Gender", gender):
```

Gender is not mandatory while create a User, so we can skip setting it entirely.
Also, **Prefer Not to Say** is the correct default IMHO.

Reply:
I'll open another one.